### PR TITLE
Add NewContact to AddressBook

### PR DESCRIPTION
**Pull request by FabianDevel**
Depends on #634 

The listeners ending with `popBackStack()` will be factorized in a  separate refactor PR

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt [hunk 1]
```diff
@@ -27,7 +27,8 @@ object ApiRoutes {
 
     fun addressBooks() = "$MAIL_API/api/pim/addressbook"
 
-    fun contacts() = "$MAIL_API/api/pim/contact/all?with=emails,details,others,contacted_times"
+    fun contact() = "$MAIL_API/api/pim/contact"
+    fun contacts() = "${contact()}/all?with=emails,details,others,contacted_times"
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:31]:
```suggestion
    fun contact() = "$MAIL_API/api/pim/contact"

    fun contacts() = "${contact()}/all?with=emails,details,others,contacted_times"
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/DetailedContactBottomSheetDialog.kt [hunk 2]
```diff
@@ -86,6 +96,11 @@ class DetailedContactBottomSheetDialog : BottomSheetDialogFragment() {
         }
     }
 
+    private fun setupSnackBar() {
+        val activity = requireActivity()
+        detailedContactViewModel.snackBarManager.setup(activity, getAnchor = { activity.findViewById(R.id.quickActionBar) })
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/DetailedContactBottomSheetDialog.kt:101]:
```suggestion
    private fun setupSnackBar() = with(requireActivity()) {
        detailedContactViewModel.snackBarManager.setup(this, getAnchor = { findViewById(R.id.quickActionBar) })
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/models/correspondent/Correspondent.kt [hunk 3]
```diff
@@ -42,5 +42,15 @@ interface Correspondent : Parcelable {
         }.uppercase()
     }
 
+    fun computeFirstAndLastName(): Pair<String, String> {
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/models/correspondent/Correspondent.kt:45]: Refactor with `computeInitials`

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/models/correspondent/Correspondent.kt (hunk 3 above)
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/data/models/correspondent/Correspondent.kt:45]: We should coordinate with iOS

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 4]
```diff
@@ -619,6 +619,19 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
     }
     //endregion
 
+    fun addContact(recipient: Recipient) = viewModelScope.launch(Dispatchers.IO) {
+        val apiResponse = ApiRepository.addContact(AddressBookController.getDefaultAddressBook().id, recipient)
+
+        val snackbarTitle = if (apiResponse.isSuccess()) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:625]:
```suggestion

        val isSuccess = ApiRepository.addContact(AddressBookController.getDefaultAddressBook().id, recipient).isSuccess()

        val snackbarTitle = if (isSuccess) {
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/DetailedContactBottomSheetDialog.kt [hunk 5]
```diff
@@ -54,6 +53,12 @@ class DetailedContactBottomSheetDialog : BottomSheetDialogFragment() {
         userAvatar.loadAvatar(navigationArgs.recipient, mainViewModel.mergedContacts.value ?: emptyMap())
         fillInUserNameAndEmail(navigationArgs.recipient, name, email)
 
+        setupListeners()
+
+        observeContacts()
+    }
+
+    private fun setupListeners() = with(binding) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/DetailedContactBottomSheetDialog.kt:61]:
```suggestion
    private fun setupListeners() = with(binding) {

```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**